services/cd-service/pkg/service/batch_test.go

```diff
@@ -145,73 +203,73 @@ func TestBatchServiceWorks(t *testing.T) {
 				t.Fatal(err)
 			}
 			for _, tr := range tc.Setup {
-				if err := repo.Apply(testutil.MakeTestContext(), tr); err != nil {
+				if err := repo.Apply(tc.context, tr); err != nil {
 					t.Fatal(err)
 				}
 			}
-			svc := &BatchServer{
-				Repository: repo,
-			}
-			resp, err := svc.ProcessBatch(
-				testutil.MakeTestContext(),
+			tc.svc.Repository = repo
+			resp, err := tc.svc.ProcessBatch(
+				tc.context,
 				&api.BatchRequest{
 					Actions: tc.Batch,
 				},
 			)
-			if err != nil {
+			if err != nil && !tc.expectedError {
 				t.Fatal(err.Error())
 			}
-			if len(resp.Results) != len(tc.Batch) {
-				t.Errorf("got wrong number of batch results, expected %d but got %d", len(tc.Batch), len(resp.Results))
-			}
 
-			// check deployment version
-			{
-				version, err := svc.Repository.State().GetEnvironmentApplicationVersion("production", "test")
-				if err != nil {
-					t.Fatal(err)
-				}
-				if version == nil {
-					t.Errorf("unexpected version: expected 1, actual: %d", version)
-				}
-				if *version != 1 {
-					t.Errorf("unexpected version: expected 1, actual: %d", *version)
-				}
-			}
-			// check that the envlock was created/deleted
-			{
-				envLocks, err := svc.Repository.State().GetEnvironmentLocks("production")
-				if err != nil {
-					t.Fatal(err)
-				}
-				lock, exists := envLocks["envlock"]
-				if !exists {
-					t.Error("lock was not created")
-				}
-				if lock.Message != "please" {
-					t.Errorf("unexpected lock message: expected \"please\", actual: %q", lock.Message)
-				}
-				_, exists = envLocks["1234"]
-				if exists {
-					t.Error("lock was not deleted")
-				}
-			}
-			// check that the applock was created/deleted
-			{
-				appLocks, err := svc.Repository.State().GetEnvironmentApplicationLocks("production", "test")
-				if err != nil {
-					t.Fatal(err)
+			if err == nil {
```

Comment:
You can avoid indenting here by just returning from that function in the line before if there is an error.